src/__test_utils__/mock-wagmi-client.ts

```diff
@@ -1,51 +1,35 @@
 /**
  * Note: https://github.com/tmm/testing-wagmi/tree/main/test was used as the starting point for this code
  */
-import { createClient, CreateClientConfig } from 'wagmi';
+import { createConfig, CreateConfigParameters, WalletClient } from 'wagmi';
 import { MockConnector } from 'wagmi/dist/connectors/mock';
 import { hardhat } from 'wagmi/chains';
-import { providers, Wallet } from 'ethers';
+import { createPublicClient, createWalletClient, http } from 'viem';
 
-type Config = Partial<CreateClientConfig> & { signer?: WalletSigner };
+type Config = Partial<CreateConfigParameters> & { walletClient?: WalletClient };
 
-export function createMockClient({ signer = getSigners()[0]!, ...config }: Config = {}) {
-  return createClient({
-    connectors: [new MockConnector({ options: { signer } })],
-    provider: () => getHardhatProvider(),
+export function createMockClient({ walletClient = getMockWalletClient(), ...config }: Config = {}) {
+  return createConfig({
+    connectors: [new MockConnector({ options: { walletClient } })],
+    publicClient: () => getPublicClient(),
     ...config,
   });
 }
 
-function getHardhatProvider() {
-  const rpcUrl = hardhat.rpcUrls.default.http[0];
-  return new providers.StaticJsonRpcProvider(rpcUrl, {
-    chainId: hardhat.id,
-    name: hardhat.name,
-    ensAddress: '0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e',
+export const getMockWalletClient = () =>
+  // Default Hardhat account
+  createWalletClient({
+    transport: http(hardhat.rpcUrls.default.http[0]),
+    chain: hardhat,
+    account: '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266',
+    key: '0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80',
+    pollingInterval: 100,
   });
-}
-
-function getSigners() {
-  const provider = getHardhatProvider();
-  return accounts.map((acc) => new WalletSigner(acc.privateKey, provider));
-}
 
-class WalletSigner extends Wallet {
-  connectUnchecked(): providers.JsonRpcSigner {
-    return (this.provider as providers.StaticJsonRpcProvider).getUncheckedSigner(this.address);
-  }
-}
-
-// Default Hardhat accounts
-const accounts = [
-  {
-    // Account #0: 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
-    privateKey: '0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80',
-    balance: '10000000000000000000000',
-  },
-  {
-    // Account #1: 0x70997970C51812dc3A010C7d01b50e0d17dc79C8
-    privateKey: '0x59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d',
-    balance: '10000000000000000000000',
-  },
-];
+export const getPublicClient = () => {
```

Comment:
Do we have this viem migration tested somewhere? I am a bit paranoid because I am not that familiar with the library.

Reply:
Yeah the Market has been running with these `viem` to `ethers` adapters since the end of June. And fyi incase it wasn't clear, these are the test utils that you are looking at. It is used here:
```ts
const wagmiClient = createMockClient();

export const render = (ui: ReactElement, options?: RenderOptions): RenderResult => {
  return baseRender(
    <BrowserRouter>
      <WagmiConfig config={wagmiClient}>{ui}</WagmiConfig>
    </BrowserRouter>,
    options
  );
};
```

Reply:
I meant it in general, but the question came out when I was looking at those tests utils, so I put it there. Sorry for confusing you.